server/src/main/java/io/crate/analyze/AnalyzedRestoreSnapshot.java

```diff
@@ -35,15 +35,30 @@
     private final String repository;
     private final String snapshot;
     private final List<Table<Symbol>> tables;
+    private final boolean includeTables;
+    private final boolean includeCustomMetadata;
+    private final List<String> customMetadataTypes;
+    private final boolean includeGlobalSettings;
+    private final List<String> globalSettings;
     private final GenericProperties<Symbol> properties;
 
     AnalyzedRestoreSnapshot(String repository,
                             String snapshot,
                             List<Table<Symbol>> tables,
+                            boolean includeTables,
+                            boolean includeCustomMetadata,
+                            List<String> customMetadataTypes,
```

Comment:
Do you think it would be worth to introduce some kind of enum for the types? It's a limited set of accepted values. Could then also use a EnumSet to enforce that the values are distinct.

In the spirit of [Parse, don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/)

Reply:
Actually I'm not sure if it is really worth creating enums for the types, we'd have to pass around these enums towards the request while we just need strings at the end. Using a set definitely makes sense.
Any concrete reasons for using enum vs string?

Reply:
> Any concrete reasons for using enum vs string?

Mostly to convey that these are not arbitrary strings and that they've been validated.
But I don't think it is a big deal in this case, also fine for me to keep it as is.